charts/cluster-api-runtime-extensions-nutanix/addons/cni/cilium/values-template.yaml

```diff
@@ -33,7 +39,17 @@ socketLB:
 envoy:
   image:
     useDigest: false
-k8sServiceHost: auto
 {{- if .EnableKubeProxyReplacement }}
 kubeProxyReplacement: true
 {{- end }}
+k8sServiceHost: "{{ trimPrefix .Cluster.Spec.ControlPlaneEndpoint.Host "https://" }}"
+k8sServicePort: "{{ .Cluster.Spec.ControlPlaneEndpoint.Port }}"
```

Comment:
I recall having issues with referencing these values, instead of using `auto`. But I might have made some mistake.

Reply:
This should always be set as per the capi control plane provider spec. auto will only work for kubeadm based control plane provider.

Reply:
The CAREN tests are passing fine with this configuration. We will have better idea when we run integration end to end tests. We will address it back if we find any issue.